### DISCUSSION TITLE
Any example code for GarageDoorOpener service

### DISCUSSION THREAD

**Discussion by rakeshnsw** [HomeSpan/HomeSpan#600] [Q&A]
Is there any example implementation of GarageDoorOpener service? I checked the documentation and there are three required characteristics: CurrentDoorState, TargetDoorState and ObstructionDetected. But there are no further details!

**Comment by HomeSpan** [comment]
See [Example 13](https://github.com/HomeSpan/HomeSpan/blob/master/docs/Tutorials.md#example-13---targetstates).

**Reply by rakeshnsw** [reply]
Great. Thanks. I believe this example doesn't include ESP GPIO related stuff! Similar to Working LED example, I'll have to include pin details to control relay to act as a switch to trigger door open/close via ESP GPIO. Right?

I may also need to include another magnetic switch or something like that to get door open/close status. But how would I integrate that with HomeSpan? I believe SpanButton method isn't suitable for this use case!

**Reply by HomeSpan** [reply]
For a magnetic switch (or any contact sensor), you should use SpanToggle() instead of SpanButton (see the API for details).

**Comment by danilkorotkov** [comment]
https://github.com/danilkorotkov/HomeSpanGate
